cctp_primitives/src/commitment_tree/mod.rs

```diff
@@ -175,10 +151,10 @@ impl CommitmentTree {
     //         otherwise returns the same as add_csw_leaf method
     pub fn add_csw(
         &mut self,
-       sc_id: &[u8; 32],
-        amount: u64,
-        nullifier: &[u8; FIELD_SIZE],
-        mc_pk_hash: &[u8; 20],
+        sc_id:      &[u8; 32],
```

Comment:
As far as I remember the sc_id bytes array should be only a valid FE serialization otherwise it will fail to be deserialized:
let sc_id = deserialize_from_buffer::<FieldElement>(sc_id_bytes).ok()?;

If you're already directly passing FEs like here:
sc_request_data: Vec<&FieldElement>,

then you could also switch sc_id type to FieldElement thus simplifying the code.